src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -17,6 +18,7 @@ trait Ledger {
 
   def executeBlock(block: Block, storages: BlockchainStorages, validators: Validators): Either[BlockExecutionError, Unit]
 
+  def prepareBlock(block: Block, storages: BlockchainStorages, validators: Validators): Either[BlockPreparationError, (BlockResult, ByteString)]
```

Comment:
maybe add a case class instead of a tuple? It's not clear what that ByteString is wihout looking in the method impl